README.md

```diff
@@ -63,6 +63,73 @@ jobs:
         commit-message: Update images digests # optional
 ```
 
+The `json` output describes the updates that `digestabot` has made and makes it
+possible to extend the functionality of the action and act on the updates in
+subsequent steps.
+
+The schema of the output is described in [`action.yml`](action.yml).
+
+```yaml
+    # Run digestabot
+    - uses: chainguard-dev/digestabot@v1
+      id: digestabot
+      with:
+        token: ${{ secrets.GITHUB_TOKEN }}
+
+    # Iterate over the updates in the `json` output
+    - shell: bash
+      run: |
+        while read -r update; do
+          updated_image=$(jq -r '.image + "@" + .updated_digest' <<<"${update}")
+
+          echo "Do something with ${updated_image} here."
+        done < <(jq -c '.updates // [] | .[]' <<<'${{ steps.digestabot.outputs.json }}')
+```
+
+You can also conditionally update images based on some criteria by reverting the
+changes made by `digestabot` and applying them yourself.
+
+```yaml
+    # Run digestabot to discover images to update. Don't create the PR.
+    - uses: chainguard-dev/digestabot@v1
+      id: digestabot
+      with:
+        token: ${{ secrets.GITHUB_TOKEN }}
+        create-pr: false
+
+    # Revert the changes made by digestabot
+    - run: |
+        git reset --hard && git clean -fd
+
+    # Iterate over the updates in the `json` output. Make the change if the image
+    # is in cgr.dev.
+    - shell: bash
+      run: |
+        while read -r update; do
+          from_image=$(jq -r '.image + "@" + .digest' <<<"${update}")
+          to_image=$(jq -r '.image + "@" + .updated_digest' <<<"${update}")
+          file=$(jq -r '.file' <<<"${update}")
+
+          if [[ "${to_image}" =~ ^cgr.dev/.+$ ]]; then
+            sed -i -e "s|$from_image|$to_image|g" "$file"
+          fi
+        done < <(jq -c '.updates // [] | .[]' <<<'${{ steps.digestabot.outputs.json }}')
+
+    # Check for any changes
+    - id: create_pr_update
+      shell: bash
+      run: |
+        echo "create_pr_update=false" >> $GITHUB_OUTPUT
+        if [[ $(git diff --stat) != '' ]]; then
+          echo "create_pr_update=true" >> $GITHUB_OUTPUT
+        fi
+
+    # Create the PR yourself
+    - uses: peter-evans/create-pull-request@v7
+      if: ${{ steps.create_pr_update.outputs.create_pr_update == 'true' }}
+      with:
+        token: ${{ secrets.GITHUB_TOKEN }}
```

Comment:
I think this is very confusing and complicated.

I prefer we have an option in digestabot that only check the images and then generate the json for you and not create any commit or pushes.

Reply:
That hypothetical option would only eliminate a few lines in this example, I think:
``` diff
-# Revert the changes made by digestabot
-- run: |
-    git reset --hard && git clean -fd
```

Is that significantly less complicated? The user will still need to create the PR themselves after choosing which updates to apply, right?

Not that I'm against putting a `noop` option in digestabot, I just wanted to document what you need to do today.

I can remove this second example for now if you'd like?